packages/main/src/Tokenizer.ts

```diff
@@ -241,31 +241,33 @@ class Tokenizer extends UI5Element {
 		}
 	}
 
-	_delete(event) {
-		if (!event.detail) { // if there are no details, the event is triggered by a click
-			this._tokenClickDelete(event, event.target);
+	_delete(e: CustomEvent) {
+		const target = e.target as Token;
+		if (!e.detail) { // if there are no details, the event is triggered by a click
+			this._tokenClickDelete(e, target);
 			return;
 		}
 
 		if (this._selectedTokens.length) {
-			this._selectedTokens.forEach(token => this._tokenKeyboardDelete(event, token));
+			this._selectedTokens.forEach(token => this._tokenKeyboardDelete(e, token));
 		} else {
-			this._tokenKeyboardDelete(event);
+			this._tokenKeyboardDelete(e, target);
 		}
 	}
 
-	_tokenClickDelete(event, token) {
+	_tokenClickDelete(e: CustomEvent, token: Token) {
 		const tokens = this._getVisibleTokens();
-		const deletedTokenIndex = token ? tokens.indexOf(token) : tokens.indexOf(event.target); // The index of the token that just got deleted
+		const target = e.target as Token;
+		const deletedTokenIndex = token ? tokens.indexOf(token) : tokens.indexOf(target); // The index of the token that just got deleted
 		const nextTokenIndex = deletedTokenIndex === tokens.length - 1 ? deletedTokenIndex - 1 : deletedTokenIndex + 1; // The index of the next token that needs to be focused next due to the deletion
 		const nextToken = tokens[nextTokenIndex]; // if the last item was deleted this will be undefined
 
 		this._handleCurrentItemAfterDeletion(nextToken);
 
-		this.fireEvent("token-delete", { ref: token || event.target });
+		this.fireEvent("token-delete", { ref: token || target });
```

Comment:
use newly created type for event details here

Reply:
Done